src/OpenTelemetry.Instrumentation.MassTransit/Implementation/MassTransitDiagnosticListener.cs

```diff
@@ -0,0 +1,83 @@
+// <copyright file="MassTransitDiagnosticListener.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Instrumentation.MassTransit.Implementation
+{
+    internal class MassTransitDiagnosticListener : ListenerHandler
+    {
+        private readonly ActivitySourceAdapter activitySource;
+        private readonly MassTransitInstrumentationOptions options;
+
+        public MassTransitDiagnosticListener(ActivitySourceAdapter activitySource, MassTransitInstrumentationOptions options)
+            : base("MassTransit")
+        {
+            this.activitySource = activitySource;
+            this.options = options;
+        }
+
+        public override void OnStartActivity(Activity activity, object payload)
+        {
+            if (!this.options.TracedOperations.Contains(activity.OperationName))
+            {
+                return;
+            }
+
+            this.activitySource.Start(activity);
+            if (activity.IsAllDataRequested)
+            {
+                activity.DisplayName = this.GetDisplayName(activity);
+            }
+        }
+
+        public override void OnStopActivity(Activity activity, object payload)
+        {
+            if (!this.options.TracedOperations.Contains(activity.OperationName))
+            {
+                return;
+            }
+
+            this.activitySource.Stop(activity);
+        }
+
+        private string GetDisplayName(Activity activity)
+        {
+            switch (activity.OperationName)
+            {
+                case OperationName.Transport.Send:
+                    return DisplayNameHelper.GetSendOperationDisplayName(this.GetTag(activity.Tags, "peer.address"));
```

Comment:
Please move these inline strings to readonly strings.

Reply:
Fixed!